src/MetadataUtility/Metadata/SupportFiles/SupportFile.cs

```diff
@@ -0,0 +1,57 @@
+// <copyright file="SupportFile.cs" company="QutEcoacoustics">
+// All code in this file and all associated files are the copyright and property of the QUT Ecoacoustics Research Group.
+// </copyright>
+
+namespace MetadataUtility.Metadata.SupportFiles
+{
+    public abstract class SupportFile
+    {
+        public string File { get; set; }
+
+        public static List<string> FindSupportFiles(TargetInformation information, string pattern)
+        {
+            List<string> logFiles = new List<string>();
+            string fileDirectory = information.FileSystem.Path.GetDirectoryName(information.Path);
+            string searchDirectory = information.FileSystem.Path.GetDirectoryName(information.Path);
+
+            int i = 0;
+
+            while (i++ < 3 && (logFiles = information.FileSystem.Directory.GetFiles(searchDirectory, pattern, SearchOption.AllDirectories).ToList()).Length() == 0)
```

Comment:
```suggestion
            const maxHeight = 3;
            while (i++ < maxHeight  && !(logFiles = information.FileSystem.Directory.GetFiles(searchDirectory, pattern, SearchOption.AllDirectories).Any())
```

Reply:
Personally, I'd just put the complexities of this into the body. You can always break or continue inside the loop body